src/View/View.php

```diff
@@ -1169,7 +1172,7 @@ protected function _evaluate(string $templateFile, array $dataForView): string
         ob_start();
 
         try {
-            include func_get_arg(0);
+            include $templateFile;
```

Comment:
Was there a reason `func_get_arg(0)` was used?

Reply:
Yes. The view variables could contain `$templateFile` and overwrite the template we include. This is extra dangerous if a controller action sets a `$templateFile` from user data as it opens an arbitrary file access vulnerability. Pretty sure this was added in the 'pre inline comment' era of CakePHP which is why there is no context with it.

Reply:
98e8aaaf682  is when this was switched to func_get_arg(0)

Reply:
We should add a comment now to clarify why the code is so.